base/interactiveutil.jl

```diff
@@ -571,11 +571,10 @@ end
 downloadcmd = nothing
```

Comment:
how about moving the check outside so that we write here
```julia
const downloadcmd = for checkcmd in (:curl, :wget, :fetch)
    if success(pipeline(`which $checkcmd`, DevNull))
        downloadcmd = checkcmd
        break
    end
end


```
and remove that from inside the `download` function ?